crates/bevy_ecs/src/storage/mod.rs

```diff
@@ -41,3 +43,71 @@ pub struct Storages {
     /// Backing storage for `!Send` resources.
     pub non_send_resources: Resources<false>,
 }
+
+/// Provides interior-mutable access to a world's internal data storages.
```

Comment:
We should link to the critical concepts here. "interior mutability" and "undefined behavior" would be very nice to link out to for newcomers to Rust who run into this in the docs.

Reply:
I agree. I imagine the rustonomicon will be a good starting point, but let me know if there are other resources you think would be worth including here.